### PR TITLE
Replace youtube username with channel id at transaction insertion time

### PR DESCRIPTION
**Pull request by nvonpentz**
Resolves https://github.com/brave-intl/internal/issues/79

Running the surveyor-frozen-report should insert all the mislabeled transactions with the correct identifiers.

This adds a test that makes an external request to the youtube data api.  Please let me know if that's not desired.

### CODE REVIEW

File: bat-utils/lib/youtube.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+'use strict'
+
+const braveHapi = require('./extras-hapi')
+
+async function getYoutubeChannelId (youtubeUsername) {
+  const youtubeApiKey = process.env.YOUTUBE_API_KEY
+  const youtubeApiBaseUri = process.env.YOUTUBE_API_BASE_URI || 'https://www.googleapis.com'
```
Comment (nvonpentz) [inline bat-utils/lib/youtube.js]: Should this be an envar? It is on publishers but I don't really see this changing.

File: bat-utils/lib/youtube.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+  let response = await braveHapi.wreck.get(`${youtubeApiBaseUri}/youtube/v3/channels?part=id&forUsername=${youtubeUsername}&key=${youtubeApiKey}`)
+  response = JSON.parse(response)
+
+  return response.items[0].id
+}
+
+module.exports = getYoutubeChannelId
```
Comment (nvonpentz) [inline bat-utils/lib/youtube.js]: Should this be it's own file or just be a function extras-utils?

File: bat-utils/lib/youtube.js (hunk 1 above)
Comment (maikelmclauflin) [reply bat-utils/lib/youtube.js]: might as well keep it as such. quicker update times

File: bat-utils/lib/youtube.js (hunk 2 above)
Comment (maikelmclauflin) [reply bat-utils/lib/youtube.js]: i'm fine with either, even going down to one function per file

File: test/eyeshade/surveyor.integration.test.js [hunk 3]
```diff
@@ -44,50 +44,66 @@ test('verify frozen occurs when daily is run', async t => {
...
+})
 
-  async function tryFreeze (dayShift, expect) {
-    await freezeOldSurveyors(debug, runtime, dayShift)
-    // beware of cursor
-    const surveyor = await querySurveyor()
-    t.is(surveyor.rowCount, 1)
-    t.is(surveyor.rows[0].frozen, expect)
-  }
+test.only('replaces invalid youtube user ids with youtube channel ids when inserting into transaction table', async t => {
```
Comment (maikelmclauflin) [inline test/eyeshade/surveyor.integration.test.js]: don't forget to remove the only

Comment (nvonpentz) [reply test/eyeshade/surveyor.integration.test.js]: ah thank you!

File: bat-utils/lib/youtube.js (hunk 2 above)
Comment (nvonpentz) [reply bat-utils/lib/youtube.js]: radical

**Pull request merged**